src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/FileSha1Step.java

```diff
@@ -0,0 +1,143 @@
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.utility.steps.fs;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.Descriptor;
+import hudson.remoting.VirtualChannel;
+import hudson.util.FormValidation;
+import jenkins.MasterToSlaveFileCallable;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.inject.Inject;
+import java.io.*;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Formatter;
+
+
+/**
+ * Compute the SHA1 of a file.
+ *
+ * @author Emanuele Zattin &lt;emanuelez@gmail.com&gt;.
+ */
+public class FileSha1Step extends AbstractStepImpl {
+    private final String file;
+
+    @DataBoundConstructor
+    public FileSha1Step(String file) throws Descriptor.FormException {
+        if (StringUtils.isBlank(file)) {
+            throw new Descriptor.FormException("can't be blank", "file");
+        }
+        this.file = file;
+    }
+
+    public String getFile() {
+        return file;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(ExecutionImpl.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "sha1";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Compute the SHA1 of a given file";
+        }
+
+        @SuppressWarnings("unused")
+        public FormValidation doCheckFile(@QueryParameter String value) {
+            if (StringUtils.isBlank(value)) {
+                return FormValidation.error("Needs a value");
+            } else {
+                return FormValidation.ok();
+            }
+        }
+    }
+
+    public static class ExecutionImpl extends AbstractSynchronousNonBlockingStepExecution<String> {
+        private static final long serialVersionUID = 1L;
+
+        @StepContextParameter
+        private transient FilePath ws;
+
+        @Inject
+        private transient FileSha1Step step;
+
+        @Override
+        protected String run() throws Exception {
+            FilePath filePath = ws.child(step.getFile());
+            return filePath.act(new ComputeSha1());
+        }
+
+        private class ComputeSha1 extends MasterToSlaveFileCallable<String> {
+            @Override
+            public String invoke(File file, VirtualChannel virtualChannel) throws IOException, InterruptedException {
+                try {
+                    if (file.exists() && file.isFile()) {
+                        return sha1(file);
+                    } else {
+                        return null;
+                    }
+                } catch (NoSuchAlgorithmException e) {
+                    e.printStackTrace();
```

Comment:
This will just end up in the agent log where no one will find it. Probably better to just let the exception pass through all the way up to `run` since it should happen very rarely.

Reply:
The signature of `invoke` only allows `IOException` and `InterruptedException`. Should I wrap `NoSuchAlgorithmException` with one of those or into a runtime exception?

Reply:
Yes, wrapping it in an `IOException` would be acceptable I guess :)